arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,331 @@
+        // Remove the last graphic added.
+        graphicsOverlay?.graphics.removeLastObject()
+        // Dismiss the table view.
+        dismiss(animated: true)
+    }
+    
+    @IBAction func doneBarButtonItemTapped(_ sender: Any) {
+        guard let feature = feature, let featureTable = featureTable else { return }
+        // Set the feature's geometry to the map point.
+        feature.geometry = mapPoint
+        // Add the feature to the feature table.
+        featureTable.add(feature) { _ in
+            // Create buffer graphics for the new feature.
+            self.delegate?.createBufferGraphics()
+            self.dismiss(animated: true)
+        }
+    }
+    
+    // MARK: Properties
+    
+    /// The geodatabase's feature table.
+    var featureTable: AGSArcGISFeatureTable?
+    /// The feature to add to the feature table.
+    var feature: AGSArcGISFeature?
+    /// The point on the map to add the feature to.
+    var mapPoint: AGSPoint?
+    /// An array of buffer sizes valid for the feature.
+    var bufferSizes: [Int]?
+    /// The graphics overlay to add the features to.
+    var graphicsOverlay: AGSGraphicsOverlay?
+    /// Indicates whether the buffer size picker is currently hidden.
+    var bufferSizePickerHidden = true
+    /// The delegate for the table view controller.
+    weak var delegate: ContingentValuesDelegate?
+    
+    /// The selected status value.
+    var selectedStatus: AGSCodedValue? {
+        didSet {
+            if let codedValueName = selectedStatus?.name {
+                // Display the selected value.
+                editRightDetail(cell: statusCell, rightDetailText: codedValueName)
+                // Reset the cell states accordingly.
+                resetCellStates()
+            }
+        }
+    }
+    
+    /// The selected protection value.
+    var selectedProtection: AGSContingentCodedValue? {
+        didSet {
+            let codedValueName = selectedProtection?.codedValue.name
+            // Display the selected protection.
+            editRightDetail(cell: protectionCell, rightDetailText: codedValueName)
+            // Reset the cell states accordingly.
+            resetCellStates()
+        }
+    }
+    
+    /// The selected buffer size.
+    var selectedBufferSize: Int? {
+        didSet {
+            if let selectedBufferSize = selectedBufferSize {
+                // Set the the buffer size attribute to the selected integer.
+                feature?.attributes["BufferSize"] = self.selectedBufferSize
+                let bufferSize = String(selectedBufferSize)
+                // Display the selected buffer size.
+                editRightDetail(cell: bufferSizeCell, rightDetailText: bufferSize)
+                // Validate the contingency.
+                validateContingency()
+            } else {
+                // If the value is nil, clear the right detail text.
+                editRightDetail(cell: bufferSizeCell, rightDetailText: " ")
```

Comment:
This 1 space string seems a bit odd. Can it be `nil`?

Reply:
When I set it to nil, the text doesn't show up after re-selection (you can reproduce by replace `" "` to `nil` and make selections for all 3 fields and then change "Status". You'll notice the text in the right detail text doesn't appear). This was just an easy fix to that solution.

Reply:
"" Will produce the same result as nil unfortunately.